### PR TITLE
new id format with random number

### PR DESCRIPTION
**Pull request by wwwdata**
it now consists of
- object_id length 2
- time length 8
- random length 6

=> total 16

### CODE REVIEW

File: spec/propertybase_id_spec.rb [hunk 1]
```diff
@@ -107,28 +101,24 @@
       end
 
       context "non existing object" do
-        let(:input_id) { "zzz456789012345678" }
+        let(:input_id) { "zzz4567890123456" }
```
Comment (leifg) [inline spec/propertybase_id_spec.rb]: Shouldn't that be only 2 z?

Comment (wwwdata) [inline spec/propertybase_id_spec.rb]: hm, the length was 17 and I just deleted a character at the end so it doesnt matter. But I could have also deleted a `z` from the front

File: lib/propertybase_id/generator.rb [hunk 2]
```diff
@@ -32,6 +24,11 @@ def host_id
       @_host_id ||= Digest::SHA1.hexdigest(Socket.gethostname).to_i(16) % PropertybaseId.max_value(2)
     end
 
+    def random_32
+      # signed 32bit integer max range
+      SecureRandom::random_number(2147483648)
```
Comment (leifg) [inline lib/propertybase_id/generator.rb]: Shouldn't the max value be 2176782335 (zzzzzz in base36)?

File: spec/propertybase_id_spec.rb (hunk 1 above)
Comment (leifg) [inline spec/propertybase_id_spec.rb]: It doesn't make any difference for the result, but for the tests it's easier to see, that the object is the only field affected

File: lib/propertybase_id/generator.rb (hunk 2 above)
Comment (wwwdata) [inline lib/propertybase_id/generator.rb]: http://ruby-doc.org/stdlib-2.2.2/libdoc/securerandom/rdoc/SecureRandom.html#method-c-random_number

hm they want a integer here and there is no `to_i` conversion of a string to base_36 directly. or is there?

File: spec/propertybase_id_spec.rb (hunk 1 above)
Comment (wwwdata) [inline spec/propertybase_id_spec.rb]: kk will change

File: lib/propertybase_id/generator.rb (hunk 2 above)
Comment (leifg) [inline lib/propertybase_id/generator.rb]: There is:

```
"zzzzzz".to_i(36)
```

Comment (wwwdata) [inline lib/propertybase_id/generator.rb]: cool, will change it

**Comment by wwwdata** [comment]
ok, pushed again with your requested changes

**Comment by leifg** [comment]
@konsti any remarks?

**Comment by wwwdata** [comment]
pushed another commit that reduces the timestamp to 7 characters as discussed with @konsti 

The overflow will now be at: `4453-04-05 17:21:36 +0200` which should give us enough time until propertybase is bankrupt and closed

**Comment by leifg** [comment]
:+1:

**Pull request merged**